### ISSUE TITLE
[Feature] Add support for executing JS actions on Page Events

### ISSUE THREAD

**Issue by ajinkyakulkarni** [appsmithorg/appsmith#6176]
## Summary

Add support for executing JS actions on page event.

Seen few use cases where users want to perform actions on page events such as 
* Page loaded
* Page unloaded
* Page deactivated
* Page activated

## Motivation

These events can simplify settings default values, preparing data before it is shown in the widgets. Many application building technologies, including JS DOM provides similar events.

## Additional Context

<img width="1362" alt="2021-07-27_12-37-05" src="https://user-images.githubusercontent.com/800578/127111766-e2947618-190a-47f8-b98b-bd935b591615.png">

**Comment by somangshu** [comment]
@ajinkyakulkarni IIUC whats missing here is 
- Page deactivated
- Page activated

**Comment by Nikhil-Nandagopal** [comment]
@ajinkyakulkarni @somangshu what does deactivated and activated mean for a page?

**Comment by ajinkyakulkarni** [comment]
@Nikhil-Nandagopal I was thinking them like focus in/out events. When you move from one page to another, these events would get fired.  Similar to Tab activated/deactivated events.

**Comment by Nikhil-Nandagopal** [comment]
@ajinkyakulkarni I guess my question is how is it different from pageLoad and unLoad? I don't know if there's a technical separation between these events

**Comment by ajinkyakulkarni** [comment]
@Nikhil-Nandagopal Yes, in our case, they could be the same if we can't technically separate between them.

**Comment by somangshu** [comment]
@ajinkyakulkarni can we close this now?

**Comment by ajinkyakulkarni** [comment]
@somangshu Do you this is implemented now or we won't be implementing it?

**Comment by somangshu** [comment]
@ajinkyakulkarni since we have `pageLoad` and `unLoad` events the above said events (activated / deactivated) are not really required, Hence this requirement can be closed?

**Comment by somangshu** [comment]
@jsartisan is the above requirement feasible?

**Comment by marks0351** [comment]
one of our users who has demoed JS editor really wishes for this enhancement. do we have any plans for its implementation?

**Comment by Nikhil-Nandagopal** [comment]
@marks0351 it's coming as part of the JS Editor. @ApekshaBhosale will be shipping this soon

**Comment by marks0351** [comment]
awesome thanks guys @Nikhil-Nandagopal @ApekshaBhosale

**Comment by shossur** [comment]
Hi Team , We need Page unload/exit event to achieve unsaved data alert for user when user moving between pages. Its on priority for US. Do you have a timeline on this?

**Comment by SG2019** [comment]
Hi, we have a similar usecase where we would like to trigger a query if the user close the page (browser).

**Comment by bharath31** [comment]
@SG2019 thanks for sharing your use case 👍🏽

**Comment by LisaPhoenixForest** [comment]
Hi!  We need the page exit events to notify our users that form field values have been modified and the edits would be discarded if not saved when leaving the page.  This is a pretty critical requirement for us as we are building an internal tool with appsmith and our admins modify end user's information.
